pkg/cloudproviders/aws/cred_manager_impl.go

```diff
@@ -0,0 +1,114 @@
+package aws
+
+import (
+	"os"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/pkg/secretinformer"
+	"github.com/stackrox/rox/pkg/sync"
+	"github.com/stackrox/rox/pkg/utils"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	cloudCredentialsKey = "credentials"
+)
+
+type awsCredentialsManagerImpl struct {
+	informer         *secretinformer.SecretInformer
+	secretFound      bool
+	mirroredFileName string
+	mutex            sync.RWMutex
+}
+
+var _ CredentialsManager = &awsCredentialsManagerImpl{}
+
+// NewCredentialsManager creates a new AWS credential manager.
+func NewCredentialsManager(
+	k8sClient kubernetes.Interface,
+	namespace string,
+	secretName string,
+	mirroredFileName string,
```

Comment:
Is the mirrored file name something we _need_  outside for later usage when initializing the SDK?

Seems like it'd be more convenient to generate it for each credentials manager using the secret name + the mirrored prefix?

Reply:
We only really need the file for AWS SDK. However we don't necessarily need it outside of the credentials manager. I exposed it to be able to pass unique file names for testing, but maybe we should rather create unique temp files without exposing the mirrored file at all.